packages/cw-pre-propose-base/src/execute.rs

```diff
@@ -0,0 +1,320 @@
+                return Err(PreProposeError::NotMember {});
+            }
+        }
+
+        let deposit_messages = if let Some(ref deposit_info) = config.deposit_info {
+            deposit_info.check_native_deposit_paid(&info)?;
+            deposit_info.get_take_deposit_messages(&info.sender, &env.contract.address)?
+        } else {
+            vec![]
+        };
+
+        let proposal_module = self.proposal_module.load(deps.storage)?;
+        let propose_messsage = WasmMsg::Execute {
+            contract_addr: proposal_module.into_string(),
+            msg: to_binary(&msg)?,
+            funds: vec![],
+        };
+
+        Ok(Response::default()
+            .add_attribute("method", "execute_propose")
+            .add_attribute("sender", info.sender)
+            .add_attribute("deposit_info", to_binary(&config.deposit_info)?.to_string())
+            .add_messages(deposit_messages)
+            .add_message(propose_messsage))
+    }
+
+    pub fn execute_update_config(
+        &self,
+        deps: DepsMut,
+        info: MessageInfo,
+        deposit_info: Option<UncheckedDepositInfo>,
+        open_proposal_submission: bool,
+    ) -> Result<Response, PreProposeError> {
+        let dao = self.dao.load(deps.storage)?;
+        if info.sender != dao {
+            Err(PreProposeError::NotDao {})
+        } else {
+            let deposit_info = deposit_info
+                .map(|d| d.into_checked(deps.as_ref(), dao))
+                .transpose()?;
+            self.config.save(
+                deps.storage,
+                &Config {
+                    deposit_info,
+                    open_proposal_submission,
+                },
+            )?;
+
+            Ok(Response::default()
+                .add_attribute("method", "update_config")
+                .add_attribute("sender", info.sender))
+        }
+    }
+
+    pub fn execute_withdraw(
+        &self,
+        deps: Deps,
+        env: Env,
+        info: MessageInfo,
+        denom: Option<UncheckedDenom>,
+    ) -> Result<Response, PreProposeError> {
+        let dao = self.dao.load(deps.storage)?;
+        if info.sender != dao {
+            Err(PreProposeError::NotDao {})
+        } else {
+            let denom = match denom {
+                Some(denom) => Some(denom.into_checked(deps)?),
+                None => {
+                    let config = self.config.load(deps.storage)?;
+                    config.deposit_info.map(|d| d.denom)
+                }
+            };
+            match denom {
+                None => Err(PreProposeError::NoWithdrawalDenom {}),
+                Some(denom) => {
+                    let balance = denom.query_balance(&deps.querier, &env.contract.address)?;
+                    if balance.is_zero() {
+                        Err(PreProposeError::NothingToWithdraw {})
+                    } else {
+                        let withdraw_message = denom.get_transfer_to_message(&dao, balance)?;
```

Comment:
do you foresee the need to ever withdraw less than the full balance? I know the withdraw is meant as a stopgap measure, but just wondering